File: labs/07/src/Style/Enumerator/FillStyleEnumerator.php
<?php

namespace App\Style\Enumerator;

use App\Shape\ShapesInterface;

class FillStyleEnumerator implements StyleEnumeratorInterface
{
    private ShapesInterface $shapes;

    public function __construct(ShapesInterface $shapes)
    {
        $this->shapes = $shapes;
    }

    public function enumStyles(callable $callback): void
    {
        $count = $this->shapes->getShapesCount();
        for ($i = 0; $i < $count; $i++) {
            $shape = $this->shapes->getShapeAtIndex($i);
            if ($shape) {
                $callback($shape->getFillStyle());
            }
        }
    }
}

File: labs/09/lab9v2/lab9v2/src/enumerator/FillStyleEnumerator.java
package enumerator;


import shape.ShapeInterface;
import style.StyleInterface;

import java.util.ArrayList;
import java.util.function.UnaryOperator;

public class FillStyleEnumerator implements StyleEnumeratorInterface {
    private ArrayList<ShapeInterface> shapes;

    public FillStyleEnumerator(ArrayList<ShapeInterface> shapes) {
        this.shapes = shapes;
    }

    @Override
    public void execute(UnaryOperator<StyleInterface> callback) {
        this.shapes.forEach(shape -> {
            callback.apply(shape.getFillStyle());
        });
    }
}
